src/api/schema/log_events/event.rs

```diff
@@ -0,0 +1,59 @@
+use crate::event;
+use async_graphql::{Enum, Object};
+use chrono::{DateTime, Utc};
+
+#[derive(Enum, Copy, Clone, PartialEq, Eq)]
+pub enum LogEventEncodingType {
+    Json,
+    Yaml,
+}
+
+#[derive(Debug)]
+pub struct LogEvent {
+    component_name: String,
+    event: event::LogEvent,
+}
+
+impl LogEvent {
+    pub fn new(component_name: &str, event: event::LogEvent) -> Self {
+        Self {
+            component_name: component_name.to_string(),
+            event,
+        }
+    }
+
+    pub fn get_message(&self) -> Option<String> {
+        Some(self.event.get("message")?.to_string_lossy())
+    }
+
+    pub fn get_timestamp(&self) -> Option<&DateTime<Utc>> {
+        self.event.get("timestamp")?.as_timestamp()
+    }
+}
+
+#[Object]
+/// Log event with fields for querying log data
+impl LogEvent {
+    /// Name of the component associated with the log event
+    async fn component_name(&self) -> &str {
+        &self.component_name
+    }
+
+    /// Log message
+    async fn message(&self) -> Option<String> {
+        self.get_message()
+    }
+
+    /// Log timestamp
+    async fn timestamp(&self) -> Option<&DateTime<Utc>> {
+        self.get_timestamp()
+    }
+
+    /// Log event as an encoded string format
+    async fn string(&self, encoding: LogEventEncodingType) -> Option<String> {
```

Comment:
I'd personally be ok with making this just `String` given we do assertions that `Event` can always serialize in other places. This would remove one bit of complexity from clients in not requiring them to handle it being null.

Reply:
IIRC, this was an open question in an earlier thread, when it wasn't obvious that serialisation is infallible. If it indeed is (and always will be, even if `LogEventEncodingType` is expanded to accommodate new/future types), then I'd certainly prefer the client to be able to rely on it being a `String!` 👍🏻

Reply:
At least JSON should be. I imagine YAML would be too, but we could verify the conditions under which it'd fail.

Reply:
Updated in 55df937dcefcd20ad5c7785af0d2e7de368f6eb1. Worst case, this will panic the thread, which is isolated to the current subscription.

I can't see any scenario where JSON/YAML wouldn't succeed, but will keep an eye out for this.

Will also merge in the changes to #6871 and drop the wrapper.